spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -695,6 +695,25 @@ trait OptimisticTransactionImpl extends TransactionalWrite
     MaterializedRowCommitVersion.throwIfMaterializedColumnNameConflictsWithSchema(metadata)
   }
 
+  /**
+   * Some features require their pre-requisite features to not only be present
+   * in the protocol but also be enabled. This method sets the flags required
+   * to enable these pre-requisite features.
+   */
+  private def getMetadataWithDependentFeaturesEnabled(metadata: Metadata): Metadata = {
+    DeltaConfigs.MANAGED_COMMIT_OWNER_NAME.fromMetaData(metadata).map { _ =>
+      // managed-commits requires ICT to be enabled as per the spec.
+      // If ICT is just in Protocol and not in Metadata,
+      // then it is in a 'supported' state but not enabled.
+      // In order to enable ICT, we have to set the table property in Metadata.
+      val ictEnablementConfigOpt =
+        Option.when(!DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetaData(metadata))(
```

Comment:
We don't need to explicitly enabled `VacuumProtocolCheckTableFeature` here as TableFeature infra takes care of enabling the requiredFeatures?

Reply:
Yes, `VacuumProtocolCheckTableFeature` does not have any metadata property that enables it. It is directly enabled/disabled in the protocol.